### PR TITLE
Java client API for log-batch

### PR DESCRIPTION
**Pull request by mparkhe**
- Add new java API
- Fixed issues with python handler for `log-batch` call

### CODE REVIEW

File: mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowProtobufMapper.java
Comment (smurching) [inline mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowProtobufMapper.java]: One thing I was kinda wondering (but I think we're good on) is if it'd make sense to accept a more general type here (e.g. like java.lang.Iterable<Metric>) since that's what the proto does. However we could always generalize the API further to accept Iterable instead of List in the future.

File: mlflow/java/client/src/test/java/org/mlflow/tracking/MlflowClientTest.java
Comment (smurching) [inline mlflow/java/client/src/test/java/org/mlflow/tracking/MlflowClientTest.java]: One observation, the `MlflowClient.logBatch` API is a little tricky to use as is given that it accepts protobuf arguments (the logic for `createMetric` is a little complex, involving using [protobuf builder](https://developers.google.com/protocol-buffers/docs/reference/java-generated#builders) APIs etc. In Python we have entity classes that make this easier and that the client APIs use (vs passing around generated protobuf classes). However in Java it seems the current pattern in MlflowClient is to use the protobuf classes, so the current API is good in that it follows the existing pattern. I also think adding a fluent API may simplify things a lot in the future too, so we might not need to worry for now. @aarondav may have thoughts / opinions here as well.

File: mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java
Comment (smurching) [inline mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java]: Nanonit: space after "params"

**Review by smurching — COMMENTED** [review]
Generally LGTM :)

Had one question about what input types we should accept - I'm not a Java expert & I don't think we need to block on that question (i.e. we could change the input types later without breaking source compatibility), so I'm ok merging as is & potentially make changes in a follow-up, but would be nice to get an answer to that question :)

File: mlflow/java/client/src/test/java/org/mlflow/tracking/MlflowClientTest.java
Comment (mparkhe) [reply mlflow/java/client/src/test/java/org/mlflow/tracking/MlflowClientTest.java]: That's a great idea. We can think of adding something like...

```
void logBatch(@NotNull String runUuid,
              @Nullable Map<String, Double> metrics,  // timestamp defaulted to sys time
              @Nullable Map<String, String> params,
              @Nullable Map<String, String> tags) {
 // create PB objects
 // call other logBatch API
}
```

File: mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowProtobufMapper.java
Comment (mparkhe) [reply mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowProtobufMapper.java]: Lemme try that ...

File: mlflow/java/client/src/test/java/org/mlflow/tracking/MlflowClientTest.java
Comment (mparkhe) [reply mlflow/java/client/src/test/java/org/mlflow/tracking/MlflowClientTest.java]: Added PR #1082 for this addition PR. Will merge in after this has been merged.

**Review by smurching — APPROVED** [review]
LGTM, thanks a bunch for the updates @mparkhe :)

**Pull request merged**
